components/brave_news/browser/feed_v2_builder.h

```diff
@@ -157,16 +158,46 @@ class FeedV2Builder : public PublishersController::Observer {
 
   void NotifyUpdateCompleted();
 
+  // Note that the |generator| will be called on a different thread.
   void GenerateFeed(UpdateSettings settings,
                     mojom::FeedV2TypePtr type,
                     FeedGenerator generator,
                     BuildFeedCallback callback);
-
-  static mojom::FeedV2Ptr GenerateBasicFeed(const FeedV2Builder& builder,
-                                            const FeedItems& items,
-                                            PickArticles pick_hero,
-                                            PickArticles pick_article);
-  static mojom::FeedV2Ptr GenerateAllFeed(const FeedV2Builder& builder);
+  void PostGenerateFeed(mojom::FeedV2TypePtr type,
+                        FeedGenerator generator,
+                        BuildFeedCallback callback,
+                        base::OnceClosure update_completed_callback);
+
+  // Threadsafe data wrappers for builder. Note that base::RefCountedData is
+  // threadsafe.
+  // TODO(sko) Some of these might not worth being refcounted, or there could
+  // be other types that should be refcounted rather than copying.
+  using RefCountedSignals = base::RefCountedData<Signals>;
```

Comment:
The same point: `RefCountedData` can't magically prevent a data race, it's just allow to safely pass and destroythe wrapped data. If UI thread modifies the data inside and a worker thread reads them at the same time we get BOOM.